### ISSUE TITLE
Audio and video accessibility accommodations

### ISSUE THREAD

**Issue by ebellempire** [omeka/Omeka#1101]
Are there any plans to provide a _text equivalent_ field (or fields) for audio and video, similar to the recently-added _alt text_ field for image files? I'm trying to figure out the best way to meet our accessibility standards for time-based media and it's a bit tricky with the available options, and I think tricky to even conceptualize in some ways. Should the item record for a video file have it's own _additional_ upload form to add an audio-only version? Should there be a way to indicate that one file is a version of another from within the files tab? Should a transcription or captions be provided as separate files or as a plain text input? I'm very comfortable coding up custom presentations at the theme level but I am very unclear about where these pieces of information belong now and where they might belong in the future. Hoping someone can provide a roadmap. I'd also be happy to participate in developing a roadmap if one doesn't yet exist.

**Comment by sharonmleon** [comment]
Thanks for your message, Erin. This is definitely something that we should work through together. 

I invite thoughts from @zerocrates, @kimisgold, and @aviannamiller.

**Comment by ebellempire** [comment]
Thanks Sharon, I look forward to hearing more. I want to add that the core issue for me is to find a solution that works with a) Items that **_are_** audio or video files (e.g. [an oral history interview](https://clevelandvoices.org/items/show/2620) where all of the Item-level metadata is about the File), as well as with b) Items that **_have_** audio or video files (e.g. in a [narrative](https://clevelandhistorical.org/items/show/1) or other compound presentation where the File-level metadata is really the only available method for description). I suspect that File metadata is not broadly-used in the grand scheme of things but – as with the alt text solution – it does seem like the right place to address this need.

**Comment by zerocrates** [comment]
Yeah this is always a challenge for us to try to accommodate both the common situation where people pretty much stick to one-file-per-item and the item describes everything, and multi-file items.

I think the question from my side is... what are we looking to enable here. "alt" is simple as it plays into a web technology that we can just plug into, but my understanding for a lot of the timed media alternative stuff is that it's more nebulous.

The route that we've gone before on linking together multiple files for this kind of alternative setup is to have the filenames match, and we look through the files on the item to see if we have a match. That's how, for example, caption files get associated with videos in a couple places. The same logic could be used for other alternatives, but I'm not sure what we'd do with more linkages like that. Captions ultimately are something you can put as a track into the video tag. Alternative audio tracks or audio versions, that's something the browsers don't really actually support yet, so we'd be looking at some other solution, or somehow presenting the connection... how much Omeka can/should be doing here is something I'm not totally clear on.

I'm fairly confident we wouldn't do a second upload within the file for another file... that feels like a lot of complexity. Anything I'm thinking about would be more in the vein of connecting "sibling" files within the same item... that could be in the form of filename matching, metadata, or some new field(s) (pick a file you're associated with, or have all the associated files have the same value in this field... I'm not sure what makes the most sense there). Again though, I'm not at all certain on what we would _do_ with such a connection once it's established.

**Comment by ebellempire** [comment]
For the sibling approach, adding a dropdown in the files list and file metadata form would be relatively straightforward. See examples below. It would mean relying more on mime type detection (or even just looking at file extensions, I guess) to provide the proper options for each file. I kind of like the idea of using multiple approaches to hopefully cover all bases and maybe improve discoverability. Having some options available directly from the Files tab in the Item form would bring more attention to the option, though it would probably mean redesigning that component to some degree or another. Adding options to the File metadata form would require very little UI by comparison and also allow for some helper text. 

<img width="511" height="145" alt="Image" src="https://github.com/user-attachments/assets/6c25a127-a0a5-424e-aeee-8d8e11ba2b18" />
<img width="296" height="503" alt="Image" src="https://github.com/user-attachments/assets/c783cb2d-17b6-42ed-9c9e-95d0e77d6229" />

**Comment by ebellempire** [comment]
For video, it might be helpful to accept two caption files, one for standard subtitles and one for [audio description](https://www.w3.org/WAI/WCAG21/Understanding/audio-description-prerecorded.html).
```
<video id="video" controls preload="metadata">
  <source src="video.mp4" type="video/mp4" />
  <track
    label="Captions"
    kind="subtitles"
    src="/path/to/subtitles.vtt"
    default />
  <track
    label="Audio Description"
    kind="subtitles"
    src="/path/to/description.vtt" />
</video>
```
Audio is more of a problem in that you cannot encapsulate the files under a standard display, the transcript is just meant to be displayed "nearby." In terms of theme design, I'd probably just provide a link to the file record where the transcript can have a full page. Something like this could probably be added to the default file markup.
```
<figure>
<audio controls>
  <source src="audio.mp3" type="audio/mpeg" />
  <p>
    Download <a download href="audio.mp3">MP3</a> or <a href="/files/show/n">view details</a>.
  </p>
</audio>
<figcaption><a href="/files/show/n">View transcript and other details</a></figcaption>
</figure>
```
Also worth noting that not all audio and video files need a textual transcription or caption treatment, but they do all need at least some kind of text-equivalent description (e.g. "a moody violin solo in a minor key" or "abstract shapes moving randomly across a canvas") so that would be the case for supporting a couple different methods.

**Comment by ebellempire** [comment]
> "...how much Omeka can/should be doing here is something I'm not totally clear on"

I think it would be fine if a lot of the implementation decisions get put on theme designers and plugin devs as long as the platform provides somewhere to store the data that's necessary to meet accessibility guidelines.

**Comment by ebellempire** [comment]
A related issue that could also be addressed using the "sibling file" approach would be to provide alternative audio and video codecs. Maybe having a secondary use case would make it worth pursuing?

**Comment by zerocrates** [comment]
The caption approach we use allows for multiple caption tracks... it's a little clunky though. The detection is just "base_file_name.vtt", so to get multiple tracks you'd need multiple files with the same original filename. We look at the title of the caption file for the label, and at the Type field for "kind." In the core this currently is only used in the lightGallery view but it (or something like it) could be extended. See https://github.com/omeka/Omeka/blob/v3.2/application/views/helpers/LightGallery.php#L169

On audio, it is allowed to use the `video` tag for audio files, and then you can use the captions tracks just as the browser supports for videos. So that might be an approach in some cases. Transcripts... I don't know if we want to provide that as a metadata field, as some specific field like alt, or something else.

On simple text-equivalent descriptions, do you know of a preferred way to expose that to assistive technologies? I can look around, but I know that there's no `alt` for these. Something as simple as an `aria-label` or `title` or something that pulls from the existing "alt" field could be something we could do simply, if that's a proper approach.

**Comment by aviannamiller** [comment]
Some techniques are linked within the [Audio-only and Video-only (Prerecorded) (Level A) SC](https://www.w3.org/WAI/WCAG21/Understanding/audio-only-and-video-only-prerecorded.html#techniques). When not incorporated directly into the video track, for instance, it looks like adding a text alternative through paragraph text (either visible or not displayed, depending on the circumstance) is one choice, if found suitable for these purposes / done in the right manner.

I can research different implementations for some actual reference points.

**Comment by kimisgold** [comment]
I recall we got a similar request from @jcmundy in https://github.com/omeka/theme-bigpicture/issues/39. @jcmundy, if you have any insights from your own use case, I'm sure it would help inform the requirements for the implementation we're looking at.

I'm trying to outline what we're potentially supporting based off discussion thus far.

* Audio
    * Field for text-based summarizing description
    * Field or upload input for text transcript with description to be displayed or linked immediately adjacent to media
* Video
    * Field for text-based summarizing description
    * Field or upload input for text transcript
    * Upload input for caption or subtitle text track
    * Upload input for audio description
      * Can be either text or audio track
    * Upload input for language audio track

Regarding the short media description, the examples I found don't seem to use a specific attribute for this information. This W3C example seems to have an `<h2>` for the video title and `<h3>` reading "Media Player". Both headings are not displayed, but remain accessible to screen readers. All the videos on [this Section 508 page of examples](https://www.section508.gov/create/synchronized-media/) seem to rely on adjacent text to contextualize and describe the media. Based off these, I'm wondering if the file's title should be sufficient? This would be putting the onus on the user to provide accurate titles, as opposed to the default file upload names.

I'm wondering if the complexity of potential options for media accessibility support would warrant its own tab in the file form, something akin to the "Advanced" tab that we have in the equivalent view in Omeka S.

**Comment by jcmundy** [comment]
The elements that you are discussing above seem like they would generally cover our needs. 

For audio descriptions for videos, the two formats that I have seen have been implemented have been:

- like YouTube, it can be handled as an alternate audio track (like a different language track) - so the video doesn't change it just has a second audio track that can be toggled to.
- in some WordPress plugins, two video options, one with the regular audio, and one with the audio description baked in, and there is a toggle between which video you want.  The user only sees one video that toggles the audio, but there are actually two videos with only different audio.

Thank you.
